src/Icon/Icon.jsx

```diff
@@ -9,28 +9,51 @@ import animationSpeeds from '../assets/animationSpeeds'
 import { default as DefaultIcon } from '@mdi/react'
 import sizes from '../assets/sizes'
 
-export const Icon = ({ name, size, color, rotate, ...props }) => {
-  const glyph = useMemo(() => {
+let currentGlyphs = { ...glyphs }
+
+export const addIcon = (name, icon) => {
+  if (currentGlyphs[name]) {
+    console.warn(`trying to redefine icon ${name}`)
+    return
+  }
+
+  currentGlyphs[name] = icon
+}
+
+export const addIcons = icons => {
+  Object.keys(icons).forEach(key => addIcon(key, icons[key]))
+}
+
+export const Icon = ({ name, size, color, rotate, face, ...props }) => {
+  const [glyph, isOldVersion] = useMemo(() => {
     if (typeof name === 'number') {
-      return String.fromCharCode(name)
+      return [String.fromCharCode(name), true]
     }
-    if (typeof name === 'string' && typeof glyphs[name] === 'number') {
-      return String.fromCharCode(glyphs[name])
+
+    if (typeof name === 'string' && typeof currentGlyphs[name] === 'number') {
+      return [String.fromCharCode(currentGlyphs[name]), true]
     }
-    return name
+
+    return [name, false]
   }, [name])
 
-  if (typeof name === 'string' && typeof glyphs[name] === 'string') {
+  const path = useMemo(() => {
+    if (name) return currentGlyphs[name]
+    return face
+  }, [name, face])
```

Comment:
Let face take precedence if the consumer specified the `face` prop:

```suggestion
  const path = useMemo(() => {
    if (face) {
      return face
    }
    return currentGlyphs[name]
  }, [name, face])
```